chat/src/main/java/net/md_5/bungee/api/chat/BaseComponent.java

```diff
@@ -665,7 +665,10 @@ void toLegacyText(StringBuilder builder)
 
     void addFormat(StringBuilder builder)
     {
-        builder.append( getColor() );
+        if ( style.hasColor() || parent != null )
```

Comment:
Why is the parent check required?

Reply:
Otherwise a constellation like `{"extra":[{"text":"b","color":"red"},{"text":"c"}], "text":"a", "color":"blue"}` would show the `c` in red instead of blue. Similar with arguments of translatables.
Unfortunaly this can introduce and not fix toLegacy conversion of BaseComponent arrays.

In new commit I added overhaul of toLegacy conversion

Reply:
Another test added for this in https://github.com/SpigotMC/BungeeCord/pull/3625/commits/dc7fecdc912dcdd700597a8f381091b2d152dde0